### PR TITLE
Enable invokes customized methods

### PR DESCRIPTION
**Pull request by henrique-gouveia**
- changed TEntity requirements on IEntityRepository interface
- added support for new repository factory definitions
- enabled invokes customized methods
- Migrate to Github Actions

### CODE REVIEW

File: test/DataQI.EntityFrameworkCore.Test/Fixtures/DbFixture.cs [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
         public DbFixture()
         {
-            Context = TestContext.NewInstance();
+            CustomerContext = TestContext.NewInstance();
+            EmployeeContext = TestContext.NewInstance();
+            ProductContext = TestContext.NewInstance();
 
             // 1. Default
-            // PersonRepository = new PersonRepository(Context);
+            // CustomerRepository = new EntityRepository<Customer, int>(Context);
```
Comment (ypedroo) [inline test/DataQI.EntityFrameworkCore.Test/Fixtures/DbFixture.cs:19]: Remove commented out code

File: test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryCustomizedMethodTes.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        {
+            var countBefore = employeeContext.Departments.Count();
+            var countExpected = ++countBefore;
+
+            var department = new Department(name);
+            employeeRepository.InsertDepartment(department);
+
+            employeeContext.SaveChanges();
+
+            Assert.True(department.Id > 0);
```
Comment (ypedroo) [inline test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryCustomizedMethodTes.cs:38]: use assertion scope on multiple assertions

File: test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryTest.cs [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+            {
+                var customer = customers.Current;
+                if (useAsyncMethod)
+                    customerRepository.DeleteAsync(customer.Id).Wait();
+                else
+                    customerRepository.Delete(customer.Id);
+
+                customerContext.SaveChanges();
+
+                Assert.False(ExistsCustomer(customer, useAsyncMethod));
```
Comment (ypedroo) [inline test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryTest.cs:217]: use assertion scope on multiple assertions

**Review by ypedroo — CHANGES_REQUESTED** [review]

File: test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryCustomizedMethodTes.cs (hunk 2 above)
Comment (henrique-gouveia) [reply test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryCustomizedMethodTes.cs:38]: This is just a way to confirm that the customer has inserted into arrange part of the unit test.

File: test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryTest.cs (hunk 3 above)
Comment (henrique-gouveia) [reply test/DataQI.EntityFrameworkCore.Test/Repository/EntityRepositoryTest.cs:217]: This is just a way to confirm that the department has deleted.

**Pull request merged**